client/components/AssignDoi/AssignDoi.js

```diff
@@ -251,8 +350,102 @@ function AssignDoi(props) {
 		handleButtonClick = handleDepositClick;
 	}
 
+	const priorPubDoi = useRef(pubData.doi);
+
+	// Re-load the preview when the Pub's DOI changes.
+	useEffect(() => {
+		if (priorPubDoi.current !== pubData.doi && status === AssignDoiStatus.Previewed) {
+			fetchPreview();
+		}
+		priorPubDoi.current = pubData.doi;
+	}, [pubData.doi, status]);
+
 	return (
 		<div className="assign-doi-component">
+			{(status === AssignDoiStatus.Previewed || status === AssignDoiStatus.Previewing) && (
+				<>
+					<Callout intent="primary" title="Review Deposit">
+						This work is being deposited as a{' '}
+						<strong>{getDepositTypeTitle(crossrefDepositRecord) || '...'}</strong> based
+						on its Connections, primary collection, or selected Content Version. Review
+						the information below, then click the "Submit Deposit" button to submit the
+						deposit to Crossref.
+					</Callout>
+					{!isStandaloneComponentDeposit(crossrefDepositRecord) && (
+						<InputField label="Content Version">
+							<Select
```

Comment:
⚠️ If possible, let's use the `{Menu, MenuItem}` from `components/Menu` here. I've been trying to switch us to this one since it has accessible keyboard interactions while the standard Blueprint `Select` sadly does not.